docs/architecture/session.md

```diff
@@ -0,0 +1,66 @@
+# Session
+
+__Session__ is the actor that encapsulates the logic of the sessions 
+associated to each TCP connections.
+
+
+
+## Actor creation and registration
+
+The creation of the session actor and its registration into the system registry are
+performed by the `connection_manager` process:
+
+```rust
+Session::create(move |ctx| {
+    // Get peer address
+    let address = stream.peer_addr().unwrap();
+
+    // Split TCP stream into read and write parts
+    let (r, w) = stream.split();
+
+    // Add stream in session actor from the read part of the tcp stream
+    Session::add_stream(FramedRead::new(r, P2PCodec), ctx);
+
+    // Create the session actor and store in its state the write part of the tcp stream
```

Comment:
// Create the session actor and store in its state the write part of the TCP stream